### PR TITLE
SG-22769 Apply rebranding to developer documentation

### PR DESCRIPTION
**Pull request by willis102**
Because the product name change on the developer docs was handled by an include, the Shotgun replacement was skipped in places where that include could not be used (i.e. inside other includes, in page titles, etc.)

This PR aims to replace all remaining instances of `Shotgun` eligible for replacement, and to update the include file to pull the trigger on the rename.

For the most part, instances of `Shotgun` in code, example logs, filesystem pats, etc. are left untouched.  In some cases, I did make replacements in example code, for instance when raising an error to the user, or in comments describing interaction with ShotGrid.

### CODE REVIEW

File: docs/en/quick-answers/developing/maya-shelf-app-launcher.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ try:
     # get the current engine (e.g. tk-maya) 
     current_engine = sgtk.platform.current_engine() 
     if not current_engine: 
-        cmds.error("Shotgun integration is not available!") 
+        cmds.error("ShotGrid integration is not available!")
```
Comment (willis102) [inline docs/en/quick-answers/developing/maya-shelf-app-launcher.md:29]: I chose to make this replacement (even though it's in code) as it's an example of an error message being presented to the user.

File: docs/en/quick-answers/developing/sgtk-script-authentication.md [hunk 2]
```diff
@@ -50,7 +50,7 @@ user = authenticator.get_user()
 
 # print "User is '%s'" % user
 
-# Tells Toolkit which user to use for connecting to Shotgun. Note that this should
+# Tells Toolkit which user to use for connecting to ShotGrid. Note that this should
```
Comment (willis102) [inline docs/en/quick-answers/developing/sgtk-script-authentication.md:53]: Similar to above, these are descriptive comments in example code, so I decided to update them as well.

**Comment by toolkitty** [comment]
[Documentation Preview](http://sg-devdocs.s3-website-us-east-1.amazonaws.com/tk-doc-generator/532f07a1ea56e17ba8d0a466c3a2b971274fcf21/index.html)

File: docs/en/quick-answers/developing/maya-shelf-app-launcher.md (hunk 1 above)
Comment (guillaume-brossard-adsk) [reply docs/en/quick-answers/developing/maya-shelf-app-launcher.md:29]: Sounds good.

**Review by guillaume-brossard-adsk — APPROVED** [review]

File: docs/en/guides/pipeline-integrations/administration/community-shared-integrations.md [hunk 3]
```diff
@@ -7,7 +7,7 @@ lang: en
 
 # Community Shared Integrations
 
-Here are projects that people in the Toolkit community have been gracious enough to share. They aren't written by us at {% include product %} and we can't vouch for them, but we'll definitely answer any questions we can about them. If you have a project you want added to the list, just email us at support@shotgunsoftware.com.
+Here are projects that people in the Toolkit community have been gracious enough to share. They aren't written by us at {% include product %} and we can't vouch for them, but we'll definitely answer any questions we can about them. If you have a project you want added to the list, please visit our [support site](https://support.shotgunsoftware.com) for help.
```
[the remark this answers is not in the document]
Comment (willis102) [reply docs/en/guides/pipeline-integrations/administration/community-shared-integrations.md:10]: I was basing this on the guidance here https://wiki.autodesk.com/pages/viewpage.action?spaceKey=SHOT&title=Shotgun+Renaming (search for support@shotgunsoftware.com)

[the remark this answers is not in the document]
Comment (shaynacohen) [reply docs/en/guides/pipeline-integrations/administration/community-shared-integrations.md:10]: That email is no longer active, and submitting requests is going to be handled differently. Instead, please direct users to  https://knowledge.autodesk.com/contact-support

[the remark this answers is not in the document]
Comment (willis102) [reply docs/en/guides/pipeline-integrations/administration/community-shared-integrations.md]: I've updated these URLs in the 12 occurrences I found.

**Comment by toolkitty** [comment]
[Documentation Preview](http://sg-devdocs.s3-website-us-east-1.amazonaws.com/tk-doc-generator/4458bff0dd6fa44165836b99255cb42612c7535d/index.html)

**Comment by toolkitty** [comment]
[Documentation Preview](http://sg-devdocs.s3-website-us-east-1.amazonaws.com/tk-doc-generator/da9c5c5e4b413848c3f39ad38da5bdb2ce34c8c2/index.html)

**Comment by toolkitty** [comment]
[Documentation Preview](http://sg-devdocs.s3-website-us-east-1.amazonaws.com/tk-doc-generator/b9ac3fa1db6a2963653262f93f46664a20be9c34/index.html)

**Pull request merged**